experimental/CliffordOrders/src/CliffordOrders.jl

```diff
@@ -0,0 +1,1056 @@
+export
+  CliffordOrder,
+  ZZCliffordOrder,
+  CliffordOrderElem,
+  ZZCliffordOrderElem,
+  clifford_order,
+  ambient_algebra,
+  coefficient_ideals,
+  even_coefficients,
+  odd_coefficients,
+  even_part,
+  odd_part,
+  center,
+  centroid,
+  quadratic_discriminant,
+  disq
+
+#############################################################
+#
+#  Datatypes
+#
+#############################################################
+
+mutable struct CliffordOrder{T, C} <: Hecke.AbstractAssociativeAlgebra{T}
```

Comment:
I think all `AbstractAssociativeAlgebra` need to be free over the base ring, to make use of some common functionality. How about making it a subtype of `Hecke.AlgAssRelOrd`?

Reply:
I would like to but Hecke.AlgAssRelOrd is not an abstract type. Is this intended?

```julia
julia> isabstracttype(Hecke.AlgAssRelOrd)
false

Reply:
Sorry, I was hallucinating. Indeed there is no abstract type. I just made them 
```
julia> supertype(Hecke.AlgAssRelOrd)
NCRing
```
since they are not necessarily free.